sdk/communication/azure-communication-sms/src/main/java/com/azure/communication/sms/SmsAsyncClient.java

```diff
@@ -102,38 +114,44 @@
      * @param from Number that is sending the message.
      * @param to A list of the recipient's phone numbers.
      * @param message message to send to recipient.
-     * @param smsOptions set options on the SMS request, like enable delivery report, which sends a report
+     * @param options set options on the SMS request, like enable delivery report, which sends a report
      * for this message to the Azure Resource Event Grid.
      * @return response for a successful send Sms request.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Iterable<SmsSendResult>> send(String from, Iterable<String> to, String message, SmsSendOptions smsOptions) {
-        SendMessageRequest request = createSendMessageRequest(from, to, message, smsOptions);
+    public Mono<Response<Iterable<SmsSendResult>>> sendWithResponse(String from, Iterable<String> to, String message, SmsSendOptions options) {
+        return sendWithResponse(from, to, message, options, null);
+    }
+
+    Mono<Response<Iterable<SmsSendResult>>> sendWithResponse(String from, Iterable<String> to, String message, SmsSendOptions options, Context context) {
         try {
             Objects.requireNonNull(from, "'from' cannot be null.");
             Objects.requireNonNull(to, "'to' cannot be null.");
-            Mono<Response<SmsSendResponse>> responseMono = withContext(context -> this.smsServiceClient.getSms().sendWithResponseAsync(request, context));
-            Response<SmsSendResponse> response = responseMono.block();
-            SmsSendResponse smsSendResponse = response.getValue();
-            List<SmsSendResult> result = convertSmsResults(smsSendResponse.getValue());
-            return Mono.just(result);
-        } catch (NullPointerException ex) {
-            return monoError(logger, ex);
+            SendMessageRequest request = createSendMessageRequest(from, to, message, options);
+            return withContext(contextValue -> {
+                if (context != null) {
+                    contextValue = context;
+                }
+                return this.smsClient.sendWithResponseAsync(request, contextValue)
+                    .flatMap((Response<SmsSendResponse> response) -> {
+                        Iterable<SmsSendResult> smsSendResults = convertSmsSendResults(response.getValue().getValue());
+                        return Mono.just(new SimpleResponse<Iterable<SmsSendResult>>(response, smsSendResults));
+                    });
```

Comment:
Why this does not need to check if smsSendResults is empty? like in line 81 above?

Reply:
Just got a comment that we should not be checking it anyway because it would be a server side issue. I will revert that.